tests/integration/targets/keycloak_group/tasks/main.yml

```diff
@@ -0,0 +1,593 @@
+      - result.end_state.realmRoles == []
+      - result.end_state.subGroups == []
+
+- name: Changing name of existing group
+  community.general.keycloak_group:
+    auth_keycloak_url: "{{ url }}"
+    auth_realm: "{{ admin_realm }}"
+    auth_username: "{{ admin_user }}"
+    auth_password: "{{ admin_password }}"
+    realm: "{{ realm }}"
+    id: "{{ result.end_state.id }}"
+    name: new-subgrp1
+    parents:
+      - "id:{{ subgrp_basegrp_result.end_state.id }}"
+  register: result
+
+- name: Debug
+  debug:
+    var: result
+
+- name: Assert that subgroup name has changed correctly
+  assert:
+    that:
+      - result is changed
+      - result.end_state != {}
+      - result.end_state.name == "new-subgrp1"
+      - result.end_state.path == "/rootgrp/new-subgrp1"
+      - result.end_state.attributes == {}
+      - result.end_state.clientRoles == {}
+      - result.end_state.realmRoles == []
+      - result.end_state.subGroups == []
+
+- name: Create a subgroup using parent name
+  community.general.keycloak_group:
+    auth_keycloak_url: "{{ url }}"
+    auth_realm: "{{ admin_realm }}"
+    auth_username: "{{ admin_user }}"
+    auth_password: "{{ admin_password }}"
+    realm: "{{ realm }}"
+    name: subgrp2
+    parents:
+      - rootgrp
+  register: result
+
+- name: Debug
+  debug:
+    var: result
+
+- name: Assert that subgroup was correctly created
+  assert:
+    that:
+      - result is changed
+      - result.end_state != {}
+      - result.end_state.name == "subgrp2"
+      - result.end_state.path == "/rootgrp/subgrp2"
+      - result.end_state.attributes == {}
+      - result.end_state.clientRoles == {}
+      - result.end_state.realmRoles == []
+      - result.end_state.subGroups == []
+
+- name: Recreate a subgroup using parent name (test idempotency)
+  community.general.keycloak_group:
+    auth_keycloak_url: "{{ url }}"
+    auth_realm: "{{ admin_realm }}"
+    auth_username: "{{ admin_user }}"
+    auth_password: "{{ admin_password }}"
+    realm: "{{ realm }}"
+    name: subgrp2
+    parents:
+      - rootgrp
+  register: result
+
+- name: Debug
+  debug:
+    var: result
+
+- name: Assert that nothing has changed
+  assert:
+    that:
+      - result is not changed
+      - result.end_state != {}
+      - result.end_state.name == "subgrp2"
+      - result.end_state.path == "/rootgrp/subgrp2"
+      - result.end_state.attributes == {}
+      - result.end_state.clientRoles == {}
+      - result.end_state.realmRoles == []
+      - result.end_state.subGroups == []
+
+## subgroup of subgroup tests
+- name: Create a subgroup of a subgroup using parent names (complete parent chain)
+  community.general.keycloak_group:
+    auth_keycloak_url: "{{ url }}"
+    auth_realm: "{{ admin_realm }}"
+    auth_username: "{{ admin_user }}"
+    auth_password: "{{ admin_password }}"
+    realm: "{{ realm }}"
+    name: subsubgrp
+    parents:
+      - rootgrp
+      - subgrp2
+  register: result
```

Comment:
Well, this basically goes back to how the keycloak API operates. The endpoint to create a subgroup is this:
```
POST /{realm}/groups/{id}/children
```
So you are principally right, one technically only needs to know the direct ancestor, but you need its ID, meaning that when giving the user the option to alternatively specify the parent by name for convenience the module must map internally from group name to group id and this is the tricky part. Actually the current `keycloak_group` module does this already, but only for top level / base groups. The way this is done is by querying this endpoint:
```
GET /{realm}/groups
```
Which returns a json structure like this:
```
[
  {
     "id": "7ec1abf0-b34f-4b06-abe3-c0db9f8565d8", 
     "name": "rootgrp",
     "path": "/rootgrp",
     "subGroups": [
         {
            "id": "2dbe4caf-e3d8-45d4-a280-690fe6bf40bd",
            "name": "new-subgrp1",
            "path": "/rootgrp/new-subgrp1",
            "subGroups": []
         },
         {
            "id": "eba34769-7c1b-492a-90fa-c37252fb7b2c",
            "name": "subgrp2",
            "path": "/rootgrp/subgrp2",
            "subGroups": [
                {
                   "id": "69196f83-cdda-4d72-be14-9c9e24d081f2",
                   "name": "subsubgrp",
                   "path": "/rootgrp/subgrp2/subsubgrp",
                   "subGroups": []
                }
            ]
         }
     ]
  }
]
```
As can be seen there sub-groups are only indirectly avaible as they are nested inside their top level parents. So yeah in theory it would be possible to search inside this tree in the ansible module code to find a matching starting point for an incomplete parent chain but this seems imho a lot of heavy lifting in potential big group structure where the sub-group name might not even be unique. So this is basically the reason I thought the best solution for (now at least) is to make it necessary to give the complete list of parents if it is all names.

To make this clear again the big issue here is the mapping between group names and the corresponding ID's. If a user happens to know the ID already (because for example it was created inside the same ansible play before and the result registered as variable) the chain can indeed be simplyfied and this is supported in the proposed code, there is also a test cased for this (Row 345ff):
```
- name: Create a subgroup of a subgroup using direct parent id (incomplete parent chain)
  community.general.keycloak_group:
    auth_keycloak_url: "{{ url }}"
    auth_realm: "{{ admin_realm }}"
    auth_username: "{{ admin_user }}"
    auth_password: "{{ admin_password }}"
    realm: "{{ realm }}"
    name: subsubsubgrp
    parents:
      - "id:{{ result_subsubgrp.end_state.id }}"
```

Reply:
Got it. I was afraid that would be the case.